### DISCUSSION TITLE
PR not created: Dependency is disabled (next)

### DISCUSSION THREAD

**Discussion by nyg** [renovatebot/renovate#14024] [Request Help] [answered]
Hello,

I've enabled Renovate for my Next.js repo and I was able to have it create a PR for minor version updates. However, there is a security update for the Next.js npm dependency and the PR is not created, even if I click on the checkbox in the Dashboard issue.

My config is the following:

<details>
<summary>renovate.json</summary>

```json
{
  "extends": [
    "group:allNonMajor"
  ],
  "schedule": [
    "at 5:00 am every saturday"
  ],
  "branchPrefix": "renovate-",
  "dependencyDashboard": true,
  "ignorePaths": [],
  "prCreation": "immediate",
  "rangeStrategy": "pin",
  "separateMultipleMajor": true,
  "stabilityDays": 3,
  "vulnerabilityAlerts": {
    "automerge": true,
    "stabilityDays": 0
  }
}
```
</details>

There are two issues I see in the logs that I haven't been able to solve:

<details>
<summary>alertPackageRules enabled set to false</summary>

```
DEBUG: GitHub vulnerability details
{
  "alerts": {
    "npm/next": {
      ">= 12.0.0, < 12.0.9": "12.0.9"
    },
    "npm/node-fetch": {
      "< 2.6.7": "2.6.7"
    }
  }
}
DEBUG: alert package rules
{
  "alertPackageRules": [
    {
      "matchDatasources": [
        "npm"
      ],
      "matchPackageNames": [
        "next"
      ],
      "matchCurrentVersion": "= 12.0.8",
      "matchFiles": [
        "package-lock.json"
      ],
      "enabled": false
    },
    {
      "matchDatasources": [
        "npm"
      ],
      "matchPackageNames": [
        "node-fetch"
      ],
      "matchCurrentVersion": "= 2.6.1",
      "matchFiles": [
        "package-lock.json"
      ],
      "enabled": false
    }
  ]
}
```
</details>

<details>
<summary>dependency is disabled</summary>

```
DEBUG: Found 2 package file(s)
INFO: Dependency extraction complete
{
  "baseBranch": "master",
  "stats": {
    "managers": {
      "github-actions": {
        "fileCount": 1,
        "depCount": 4
      },
      "npm": {
        "fileCount": 1,
        "depCount": 9
      }
    },
    "total": {
      "fileCount": 2,
      "depCount": 13
    }
  }
}
DEBUG: Dependency is disabled (next)(dependency="next")
```
</details>

I've tried to force `enabled` to true in `vulnerabilityAlerts` as well as in `npm` and `packageRules` for all packages but to no avail.

Thanks a lot for your help!
nyg

Public repo: https://github.com/nyg/dependabot-vuln-viewer

**Comment by rarkins** [comment, marked as answer]
These are transitive dependencies, and possibly not fixable. I'll look into it. We should ideally remove them from the dashboard if we can't fix.

![image](https://user-images.githubusercontent.com/6311784/152639297-2fee7437-c31d-4c53-8ff6-ac4c0097238d.png)

**Reply by nyg** [reply]
Hi, thanks for the reply! For node-fetch, yes it's a transitive dependency. But I don't understand why next would be one too? It is specified in the package.json.

**Reply by rarkins** [reply]
Agreed, a fix is needed for this

**Reply by nyg** [reply]
Aha okay, thanks for looking into it :). Is there already an issue/PR I can follow to know when this will be fixed?

EDIT: found #14025 and #14082

**Reply by rarkins** [reply]
That first PR is somewhat of a bandaid, just to remove impossible remediations from the dashboard so as not to confuse people.

The fuller fix is to enable transitive remediation for npm >= 7.
